Source/playerdat.cpp

```diff
@@ -152,11 +155,122 @@ void ReloadExperienceData()
 	}
 }
 
+void LoadClassAttributes(std::string_view classPath, ClassAttributes &out)
+{
+	const std::string filename = StrCat("txtdata\\classes\\", classPath, "\\attributes.tsv");
+	tl::expected<DataFile, DataFile::Error> dataFileResult = DataFile::load(filename);
+	if (!dataFileResult.has_value()) {
+		DataFile::reportFatalError(dataFileResult.error(), filename);
+	}
+	DataFile &dataFile = dataFileResult.value();
+
+	enum class ClassAttributeColumn {
+		Attribute,
+		Value,
+		LAST = Value
+	};
+	std::array<ColumnDefinition, enum_size<ClassAttributeColumn>::value> columns;
+	if (tl::expected<void, DataFile::Error> result
+	    = dataFile.parseHeader<ClassAttributeColumn>(columns.data(), columns.data() + columns.size(),
+	        [](std::string_view name) -> tl::expected<ClassAttributeColumn, ColumnDefinition::Error> {
+		        if (name == "Attribute")
+			        return ClassAttributeColumn::Attribute;
+		        if (name == "Value")
+			        return ClassAttributeColumn::Value;
+		        return tl::make_unexpected(ColumnDefinition::Error::UnknownColumn);
+	        });
+	    !result.has_value()) {
+		DataFile::reportFatalError(result.error(), filename);

```

Comment:
With the way you're currently handling values you'll want to report an error here if both columns are present but not in the expected order.

Reply:
There is no expected order, it's an `unordered_map`

Reply:
Ah, you mean columns, yeah.
I've changed it to simply `skipHeader()`